### ISSUE TITLE
Improve the Tags suggestions delayed audible messages (speak)

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#1510]
The Tags suggestion accessible search implemented in #1141  makes use of lodash `throttle` see https://github.com/WordPress/gutenberg/blob/6e6fac6b98135e6047810e6dcc470ccf5af2596e/components/form-token-field/index.js#L191

Seems to me it should use `debounce` instead, as `throttle` just delays the call to `wp.a11y.speak()` that will run after 1 second, regardless the user is still typing or not.

To test:
- focus the Tags field and start slowly typing a non sense term
- keep typing
- observe in your browser inspector how the aria-live region gets updated with the speak message while you're still typing

In the PR for the Inserter blocks search results #1508 I'm proposing to use `debounce` and seems to me it works better, as our goal should be delivering the message only when users stop typing.
One thing to consider could be making a decision about the delay: maybe 1000ms is a bit too much.
On the other hand, not all users type at the same speed and some users with motor impairments type very slowly, so any fixed value will always be an educated guess or assumption.

**Comment by afercia** [comment]
I've prepared a short video (16 seconds!) to illustrate what's the current behavior:
https://cloudup.com/c99h7YOTAmz
Notice how the `No results.` message quickly appears multiple times while I'm still typing.
This is not the ideal behavior, messages should be sent to users only when they stop typing.
